search/includes/classes/class-search.php

```diff
@@ -1955,4 +1958,24 @@ public function log_ep_invalid_response( $request, $query ) {
 
 		$this->logger->log( 'warning', 'search_query_failure', $message );
 	}
+
+	public function maybe_change_index_version() {
+		$indexables = $this->indexables->get_all();
+
+		foreach ($indexables as $indexable) {
+			$query_force_version_argument = sprintf( self::QUERY_FORCE_VERSION_PATTERN, $indexable->slug );
+
+			if ( ! isset( $_GET[ $query_force_version_argument ] ) ) {
+				continue;
+			}
+
+			$normalized_version = $this->versioning->normalize_version_number( $indexable, $_GET[ $query_force_version_argument ] );
```

Comment:
`set_current_version_number` is doing `normalize_version_number` as well. The reason I explicitly called it here, is that It felt safer with regards to working directly with user input. This way we make sure we always validate the input even if the implementation is changed in `set_current_version_number`. 

But then again this might be a bit of overkill and we could drop these 4 lines making the code simpler.

What do you think?

Reply:
I think it's fine to keep it explicit like that.